docs/guide/custom-ui.md

```diff
@@ -438,25 +450,139 @@ module.exports = {
-* Overriding the same `action` is possible (the last wins).
-* You may mark the action as "dangerous" via an optional property `modifiers: [ 'danger' ]` (see the 'Archive' and 'Unpublish' menu items).
-* An additional optional boolean property `manuallyPublished` is supported. When set to true, the custom menu item is available only for document types that do not have the `autopublish: true` or `localized: false` options set.
-* The `conditions` property is optional. It takes an array of one or more strings specifying conditions that all must be satisfied to determine if the action can be run on the current doc. Valid values are: 'canPublish', 'canEdit', 'canDismissSubmission', 'canDiscardDraft', 'canLocalize', 'canArchive', 'canUnpublish', 'canCopy', 'canRestore'. To go beyond these, see the more flexible `if` and `moduleIf` features below.
-* The optional `moduleName` property can be used to override the `moduleName` prop passed to the modal. By default, it will be the name of the piece type module corresponding to the individual piece, or `@apostrophecms/page` in the case of pages.
-* The `if` property takes an object that works like a MongoDB query criteria object, in a limited way: each property must match the corresponding property of the document. The `$or`, `$and` and `$ne` operators are supported as in MongoDB, along with dot notation to match nested properties. Other MongoDB query features are not supported at this time.
-* The `moduleIf` property works like `if`, but it matches properties of the module rather than the document. Note this extends only to properties passed down to the browser via the `getBrowserData` method of the module in question. You can expose new properties via this method by using the `extendMethods` feature, [as mentioned here](/reference/modules/piece-type.html#getbrowserdata-req).
-* For backward compatibility, this method can also be called with the `moduleName` passed as the first argument and the object as the second, but this is discouraged.
-:::
+Alternatively, you can create a context menu item that emits an event instead of opening a modal:
+
+<AposCodeBlock>
+
+```js
+module.exports = {
+  extend: '@apostrophecms/piece-type',
+  options: {
+    label: 'Article',
+    pluralLabel: 'Articles'
+  },
+  init(self) {
+    self.apos.doc.addContextOperation({
+      context: 'update',
+      action: 'refresh-related-content',
+      label: 'Refresh Related Content',
+      type: 'event',
+      if: {
+        type: self.__meta.name
+      }
+    });
+```
+  <template v-slot:caption>
+    modules/article/index.js
+  </template>
+</AposCodeBlock>
+
+
+When a user clicks on an event-type context menu item, it triggers an event with the same name as the `action` property. You can listen for this event elsewhere in your code:
+
+<AposCodeBlock>
+
+``` javascript
+// Listen for the event elsewhere in your code or another module
+self.apos.bus.$on('refresh-related-content', (data) => {
```

Comment:
Would be better to remove the `self` since we frontend and apos is attached to window

```javascript
apos.bus.$on('refresh-related-content', (data) => {
```

Reply:
Changed

Reply:
This applies to the `self.notify` also, correct?

Reply:
Yes, it's apos.notify on frontend side.